static/src/javascripts/projects/common/modules/commercial/commercial-feature-policies.js

```diff
@@ -68,16 +68,15 @@ define([
         }
     };
 
-    policies.membershipMessages = function () {
-        if (!detect.adblockInUseSync() &&
-            detect.getBreakpoint() !== 'mobile' &&
-            config.page.contentType === 'Article' &&
-            !userFeatures.isPayingMember()
-        ) {
-            return {
-                membershipMessages : true
-            };
-        }
+    policies.checkWeCanShowMembershipMessages = function () {
```

Comment:
i'm pretty uneasy about naming things after their implementation, rather than their role. if promises are replaced, this function will presumably still be async (maybe even `then`able) but not a promise. `checkX.then` seems so much easier to understand, to my mind, than `xPromise.then`?